plugin/client_rpc.go

```diff
@@ -62,12 +62,38 @@ type apiRPCServer struct {
 	impl API
 }
 
+// ErrorString is a fallback for sending unregistered implementations of the error interface across
+// rpc. For example, the errorString type from the github.com/pkg/errors package cannot be
+// registered since it is not exported, but this precludes common error handling paradigms.
+// ErrorString merely preserves the string description of the error, while satisfying the error
+// interface itself to allow other registered types (such as model.AppError) to be sent unmodified.
+type ErrorString struct {
+	Err string
+}
+
+func (e ErrorString) Error() string {
+	return e.Err
+}
+
+func encodableError(err error) error {
+	if err == nil {
```

Comment:
I would prefer here to use "idiomatic" error handling. for me is easier to follow the code:

```go
	if err == nil {
		return nil
	}
	if _, ok := err.(*model.AppError); ok {
		return err
	}
	return &ErrorString{
		Err: err.Error(),
	}
```

Because this way I can read lines and forget about that line. I mean, if err is nil, I don't need to see anything more, but if not, I don't care about this line anymore, anything else is about handling any other case. For elseif you have to think about the if clause and then apply the new if clause, and for the last else, you have to think about any if/elseif clauses before.

Reply:
Thank you for the reminder! Yes, I agree that your example is more idiomatic. It still feels "weird", but when I review code written by the Go authors, I see this pattern frequently, so I'm happy to get on the bandwagon.

Fixed!